Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,52 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
-            var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+        private static void Test(string args, int expected)
+        {
+            var C = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
+            Console.WriteLine($"{args} : {C}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
-        }
-    }
+        public static int FindDigit(string equation)
+        {   
+            string C = equation.Split('=')[1];
+            string A = equation.Split('*')[0];
+            string B = equation.Split('*')[1].Split('=')[0];
+            float compResult = -1;
+            string givenResult = "";
+            
+
+            if(A.IndexOf('?') > -1){
+                compResult = ((float)Int32.Parse(C))/Int32.Parse(B);
+                givenResult = A;
+            }
+            else if(B.IndexOf('?') > -1){
+                compResult = ((float)Int32.Parse(C))/Int32.Parse(A);   
+                givenResult = B;
+            }
+            else if(C.IndexOf('?') > -1){
+                compResult = Int32.Parse(A)*Int32.Parse(B);
+                givenResult = C;
+            }
+
+           // computed result is not integer, no integer solution possible
+            if(compResult != (int)compResult){
+                return -1;
+            }
+
+            string res = compResult.ToString();
+
+            if(res.Length != givenResult.Length){
+                // length mismatch
+                return -1;
+            }
+            else{
+
+             int index=givenResult.IndexOf('?');
+             return (int)res[index]-48;

```

Comment:
please provide comments where necessary to understand what is being done. If possible try to create small private methods and incorporate logic inside them so that is code more readable.